numpy/random/pcg32.pyx

```diff
@@ -128,148 +110,55 @@ cdef class PCG32:
-
-        cdef const char *name = "BitGenerator"
-        self.capsule = PyCapsule_New(<void *>&self._bitgen, name, NULL)
-
-    # Pickling support:
-    def __getstate__(self):
-        return self.state
-
-    def __setstate__(self, state):
-        self.state = state
-
-    def __reduce__(self):
-        from ._pickle import __bit_generator_ctor
-        return __bit_generator_ctor, (self.state['bit_generator'],), self.state
-
-    def random_raw(self, size=None, output=True):
-        """
-        random_raw(self, size=None)
-
-        Return randoms as generated by the underlying BitGenerator
-
-        Parameters
-        ----------
-        size : int or tuple of ints, optional
-            Output shape.  If the given shape is, e.g., ``(m, n, k)``, then
-            ``m * n * k`` samples are drawn.  Default is None, in which case a
-            single value is returned.
-        output : bool, optional
-            Output values.  Used for performance testing since the generated
-            values are not returned.
-
-        Returns
-        -------
-        out : uint or ndarray
-            Drawn samples.
-
-        Notes
-        -----
-        This method directly exposes the the raw underlying pseudo-random
-        number generator. All values are returned as unsigned 64-bit
-        values irrespective of the number of bits produced by the PRNG.
-
-        See the class docstring for the number of bits returned.
-        """
-        return random_raw(&self._bitgen, self.lock, size, output)
-
-    def _benchmark(self, Py_ssize_t cnt, method=u'uint64'):
-        return benchmark(&self._bitgen, self.lock, cnt, method)
-
-    def seed(self, seed=None, inc=0):
-        """
-        seed(seed=None, inc=0)
-
-        Seed the generator.
-
-        This method is called at initialization. It can be called again to
-        re-seed the generator.
-
-        Parameters
-        ----------
-        seed : int, optional
-            Seed for ``PCG64``. Integer between 0 and 2**64-1.
-        inc : int, optional
-            Increment to use for PCG stream. Integer between 0 and 2**64-1.
-
-        Raises
-        ------
-        ValueError
-            If seed values are out of range for the PRNG.
-        """
-        ub = 2 ** 64
-        if seed is None:
-            try:
-                seed = <np.ndarray>random_entropy(2)
-            except RuntimeError:
-                seed = <np.ndarray>random_entropy(2, 'fallback')
-            seed = seed.view(np.uint64).squeeze()
-        else:
-            err_msg = 'seed must be a scalar integer between 0 and ' \
-                      '{ub}'.format(ub=ub)
-            if not np.isscalar(seed):
-                raise TypeError(err_msg)
-            if int(seed) != seed:
-                raise TypeError(err_msg)
-            if seed < 0 or seed > ub:
-                raise ValueError(err_msg)
-
+        # Seed the _bitgen
+        val = self._seed_seq.generate_state(1, np.uint64)
```

Comment:
While I think we've all more or less agreed that we should just remove `PCG32`, I will just note that the scheme for seeding `PCG32` should be the same as `PCG64`: grab both state and inc from the seed sequence.

Reply:
ok